profiler/profiler.go

```diff
@@ -604,7 +615,11 @@ func pollProfilerService(ctx context.Context, a *agent) {
 	debugLog("Cloud Profiler Go Agent version: %s", version.Repo)
 	debugLog("profiler has started")
 	for i := 0; config.numProfiles == 0 || i < config.numProfiles; i++ {
-		p := a.createProfile(ctx)
+		p, err := a.createProfile(ctx)
+		if err != nil {
+			debugLog("Cloud Profiler Go Agent encountered permanent error, profiling will be disabled.")
```

Comment:
Outside of this CL: maybe we should make debugLog() add "Cloud Profiler Go Agent: " as a prefix automatically. Need to handle multi-line messages.